prow/git/v2/client_factory.go

```diff
@@ -114,12 +123,8 @@ func defaultClientFactoryOpts(cfo *ClientFactoryOpts) {
 		cfo.Host = "github.com"
 	}
 	if cfo.CacheDirBase == nil {
-		switch runtime.GOOS {
-		case "linux":
-			cfo.CacheDirBase = utilpointer.StringPtr("/var/tmp")
-		default:
-			cfo.CacheDirBase = utilpointer.StringPtr("")
-		}
+		// If we do not have a place to put cache, put it in temp dir.
+		cfo.CacheDirBase = defaultTempDir()
```

Comment:
I can't see any value in moving this to its own function.

Reply:
This is in its own function because we want to use it twice. Once to default cacheDirBase (when not provided by the user) The other time to set the location where repos will be copied (Always no matter what the cache dir base is)